docs/source/data/subscriptions.mdx

```diff
@@ -47,139 +49,148 @@ subscription PostFeed {
+3. Next, in order to set up both the HTTP and subscription servers, we need to first create an `http.Server`. Do this by passing your Express `app` to the `createServer` function, which we imported from the `http` module:
 
-    ```javascript:title=index.js
+    ```ts:title=index.ts
     // This `app` is the returned value from `express()`.
     const httpServer = createServer(app);
     ```
 
-4. Create an instance of `GraphQLSchema` (if you don't already).
+4. Create an instance of `GraphQLSchema` (if you haven't already).
 
     > If you already pass the `schema` option to the `ApolloServer` constructor (instead of `typeDefs` and `resolvers`), you can skip this step.
 
-    The `SubscriptionServer` (which we'll instantiate next) doesn't take `typeDefs` and `resolvers` options. Instead, it takes an executable `GraphQLSchema`. We can pass this `schema` object to both the `SubscriptionServer` and `ApolloServer`. This way, we make sure that the same schema is being used in both places.
+    The subscription server (which we'll instantiate next) doesn't take `typeDefs` and `resolvers` options. Instead, it takes an executable `GraphQLSchema`. We can pass this `schema` object to both the subscription server and `ApolloServer`. This way, we make sure that the same schema is being used in both places.
 
-    ```javascript:title=index.js
+    ```ts:title=index.ts
     const schema = makeExecutableSchema({ typeDefs, resolvers });
     // ...
     const server = new ApolloServer({
       schema,
     });
     ```
 
-5. Create the `SubscriptionServer`.
-
-    ```javascript:title=index.js
-    const subscriptionServer = SubscriptionServer.create({
-       // This is the `schema` we just created.
-       schema,
-       // These are imported from `graphql`.
-       execute,
-       subscribe,
-    }, {
-       // This is the `httpServer` we created in a previous step.
-       server: httpServer,
-       // Pass a different path here if your ApolloServer serves at
-       // a different path.
-       path: '/graphql',
-    });
+5. Create a `WebSocketServer` to use as your subscription server.
+
+    ```ts:title=index.ts
+      // Creating the WebSocket server
+      const wsServer = new WebSocketServer({
+        // This is the `httpServer` we created in a previous step.
+        server: httpServer,
+        // Pass a different path here if your ApolloServer serves at
+        // a different path.
+        path: '/graphql',
+      });
+
+      // Hand in the schema we just created and have the
+      // WebSocketServer start listening.
+      useServer({ schema }, wsServer);
     ```
 
-6. Add a plugin to your `ApolloServer` constructor to close the `SubscriptionServer`.
+6. Add [a plugin](../integrations/plugins) to your `ApolloServer` constructor to close the `WebSocketServer`:
```

Comment:
Oh huh, I need to make sure to find the right way to do this in my example. I don't know if this is the right API for graphql-ws.

Reply:
Fixed this in the example in https://github.com/apollographql/docs-examples/pull/40

Note that this involved not just draining the subscription server, but also draining the main HTTP server! Also had to reorder some stuff so that things got defined in the right order.

Reply:
Oooh thank you for updating the example! I'll update the set up instructions in this article to follow suit. :)